### ISSUE TITLE
Duplicated frames when rendering timelapse if "change timepoint to 0"

### ISSUE THREAD

**Issue by nelas** [bene51/3Dscript#26]
Hello,

I've noticed that if I render a timelapse using the code below (based on the zebrafish example in the [gallery](https://bene51.github.io/3Dscript/gallery.html)), I get many duplicated frames in the resulting animation. In this particular case there were ~30 duplicated frames out of 150 frames.

```
At frame 0:
- change timepoint to 0

From frame 0 to frame 150 change timepoint to 150
```

However, if I simply change the timepoint at frame 0 to `1` instead of `0`, the resulting animation has no duplicated frames (except for the first one) and works as expected.

```
At frame 0:
- change timepoint to 1

From frame 0 to frame 150 change timepoint to 150
```

Not sure if it's a bug, but maybe it is worth mentioning in the wiki or gallery? I ran into this when I tried to compare two different renderings frame-by-frame and could not because there were random duplicated frames in the animations.

Thanks!
Bruno

**Comment by bene51** [comment]
Hi Bruno,

thanks for letting me know. May I ask you how many timepoints your timelapse dataset has? Exactly 150?

Best wishes,
Bene

**Comment by nelas** [comment]
Hi Bene, oh yes, this dataset has exactly 150 timepoints. Shown as `t:1/150` in the ImageJ dialog.

Best,
Bruno

**Comment by bene51** [comment]
Hi Bruno,

there was indeed a rounding issue, which is hopefully solved by https://github.com/bene51/3Dscript/commit/e1774349c146a7fafaa1c72bcc1f7bf0e20f4616.

I uploaded a new version to the Fiji/ImageJ update site, you should get the changes when you update your ImageJ next time.

Additionally, there's no a "Dry run..." command in the editor's 'Run' menu, which will not render, but just show you all the settings for all the frames, this makes it hopefully easier in the future to pinpoint issues.

BTW: Here's a way I created a 150 timepoint stack to investigate your issue:
```
imp = ij.IJ.createImage("", 256, 256, 100 * 150, 8);
imp.setDimensions(1, 100, 150);
imp.setOpenAsHyperStack(true);
for(int t = 0; t < 150; t++) {
	int idx = imp.getStackIndex(1, 1, t + 1);
	ip = imp.getStack().getProcessor(idx);
	ip.setValue(255);
	ip.setFont(new java.awt.Font("Helvetica", java.awt.Font.PLAIN, 24));
	ip.drawString("TP " + t, 0, 128);
}
imp.show();
```
You can copy&paste this into the Fiji script editor (>File>New>Script...), choose 'Beanshell' from the Language menu and click on 'Run'.

It would be great if you could verify that the updated version resolves your original issue.

Thanks and best wishes,
Bene

**Comment by nelas** [comment]
Hi Bene,

I just tested and it is fixed, I no longer get duplicated frames as described above.

From the testing I noticed that to get an animation with exactly one frame per timepoint (given the initial 150 timepoints), I need to:

- Set `change timepoint to 1` and not `0` (otherwise frames 0 and 1 are duplicated)
- Set `From frame 0 to frame 149` and not `150` (otherwise the final stack will have 151 frames in total - one duplicated timepoint somewhere in the middle).

That would be:

```
At frame 0:
- change timepoint to 1

From frame 0 to frame 149 change timepoint to 150
```

Thanks for the reply and for developing 3Dscript! It has been really helpful for rendering my data.

Best,
Bruno
